Dockerfile

```diff
@@ -0,0 +1,48 @@
+### STAGE 1: Build the AngularJS app ###
+
+FROM node:10-alpine as build-stage
+
+RUN apk --no-cache add git
+
+WORKDIR /spa
+
+COPY package.json /spa/
+
+RUN npm install
+
+COPY . .
+
+# Production mode build
+RUN npm run build:prod
+
+
+### STAGE 2: Add Nginx for hosting the AngularJS app ###
+
+FROM nginx:1.19.1-alpine
+
+# Removes the default nginx html files
+RUN rm -rf /usr/share/nginx/html/*
+
+# Copy the bundle
+COPY --from=build-stage /spa/dist /usr/share/nginx/html
+
+# Copy the default nginx.conf
+COPY --from=build-stage /spa/nginx.conf /etc/nginx/conf.d/default.conf
+
+VOLUME ["/usr/share/nginx/html/env"]
+
+EXPOSE 80
+
+# Copy .env file and shell script to container
+WORKDIR /usr/share/nginx/html
+COPY ./scripts/generate_env.sh .
+COPY .env .
```

Comment:
No. We do not rely on the .env file, we rely on the variables from Docker run which is a requirement from the devops team:

```sh
docker run --env OPENPAAS_BASE_URL=foo --env-file ./env.list openpaas-suite/contacts-spa
```

Then in the script below, you must replace the value in the JSON file.